Add docs on creating and accessing AWS accounts

richardTowers opened the pull request.

Commit:
Add page on accessing AWS accounts

Commit:
trello-XU793Z54: Add documentation on creating new accounts

Commit:
Add guidance on how to add new users to AWS

Commit:
Fix broken link

Commit:
Correct link to form in create-aws-accounts

philandstuff commented on source/manuals/access-aws-accounts.html.md.erb:
```diff
@@ -0,0 +1,154 @@
+the base account it is up to the owners of the target accounts to grant the
+user access to roles within the account.
+
+To add a user to the `gds-users` base account fill in [this form](https://gds-request-an-aws-account.cloudapps.digital/).
```
we generally do reference-style markdown links in this repo. That is, links like this:

```
You can access [foo][] by doing steps [bar and baz][alternate-id].

[foo]: https://example.com/foo
[alternate-id]: https://example.com/doing-steps-bar-and-baz
```

Aside from being consistent, I find this helps with editing the text without the links getting in the way of reading whole sentences.

philandstuff commented on source/manuals/access-aws-accounts.html.md.erb:
```diff
@@ -0,0 +1,154 @@
+    </details>
+  </li>
+  <li>
+    Fill in the account number and role you're switching to
```
is it worth mentioning account aliases as well here?

philandstuff commented on source/manuals/access-aws-accounts.html.md.erb:
```diff
@@ -0,0 +1,154 @@
+ask the user for an MFA token. To work with these tools you can create a token using
+[STS](https://docs.aws.amazon.com/STS/latest/APIReference/Welcome.html).
+
+Using [`awscli`](https://aws.amazon.com/cli/), run:
```
I would prefer to recommend `aws-vault` instead of directly calling `aws sts assume-role` here.

philandstuff commented on source/manuals/create-aws-accounts.html.md.erb:
```diff
@@ -0,0 +1,34 @@
+* Minimize the impact of security breaches
+* Isolate audit data in separate accounts
+
+See [the AWS multi-account guidance](https://aws.amazon.com/answers/account-management/aws-multi-account-security-strategy/).
```
ditto on reference-style links

philandstuff reviewed (commented):
Really happy to see this change 👍 

Some minor comments inline

(FYI: I'm on leave all next week; don't feel you have to wait for me to come back to get this PR merged)

philandstuff commented:
Also, you got merge conflicts 💥

soupdragon99 committed:
Merge branch 'master' into trello-XU793Z54-aws-account

soupdragon99 commented:
I've fixed the merge conflicts now.

soupdragon99 committed:
Tech writer edits

soupdragon99 commented on source/manuals/access-aws-accounts.html.md.erb:
```diff
@@ -11,20 +11,17 @@ account_list:
+
+## Access the AWS Console
 
 <ul>
```
We had UR feedback about using these drop down widgets, that they're incompatible with screen readers and can break the left-hand nav' in the tech doc format.

soupdragon99 committed:
Tech writer edits to creating AWS accounts

Commit:
trello-XU793Z54: Fixes to AWS accounts docs

* fix a couple of broken links
* move the list of lists of accounts out of the <details> block so it's more visible
* mentions account aliases in the switch role bit (as per Phil's suggestion)

Commit:
trello-XU793Z54: Expand documentation on setting up accounts

Commit:
trello-XU793Z54: Mention aws-vault as an alternative

Would probably be better to standardise on using aws-vault, but since
I've never even used it I don't feel super qualified on documenting it.
